### PR TITLE
Apa toast rework2

### LINKED ISSUE #1000: Rewrite the toasts implementation for crossplatform
The current implementation of toasts in chatterino is currently not that good since it doesn't support very basic features and isn't crossplatform which chatterino strives to be.

### LINKED ISSUE #1435: Streamer profile picture in the live notification in chatterino is broken
**Describe the bug**
Since 2.1.5 whenever a streamer goes live it's just the standard "broken" profile picture, it works flawlessly in 2.1.4

**To reproduce**
Install 2.1.5, go to settings and then "notifications", activate "Enable Toasts" and add a channel in the tab "Channel going live"

**Screenshots**
2.1.4 

2.1.5 

**Chatterino version**
Chatterino 2.1.5 (commit c6e1a1de2)

**Operating system**
Windows 10 Pro Version 1909 OS Build 18363.476

**Additional information**
I also tried the latest nightly versions, it's also broken there.

### PR DESCRIPTION
**Pull request by apa420**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description
Toasts with crossplatform support.

Fixes #1000
Fixes #1435

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## Unversioned
 
+- Major: Crossplatform toast notification support.
```
Comment (zneix) [inline CHANGELOG.md:5]:
```suggestion
- Major: Added Cross-platform toast notification support. (#1985)
```

File: src/singletons/Toasts.cpp [hunk 2]
```diff
@@ -64,152 +42,108 @@ QString Toasts::findStringFromReaction(
...
 {
-private:
-    QString channelName_;
-    Platform platform_;
-
-public:
-    CustomHandler(QString channelName, Platform p)
-        : channelName_(channelName)
-        , platform_(p)
+    QString bottomText = "";
```
Comment (zneix) [inline src/singletons/Toasts.cpp:59]:
```suggestion
    QString bottomText;
```

File: src/singletons/Toasts.cpp [hunk 3]
```diff
@@ -64,152 +42,108 @@ QString Toasts::findStringFromReaction(
...
+            [channelName, bottomText, callback, this](auto result) -> Outcome {
+                const auto data = result.getData();
+
+                QPixmap avatar;
+                avatar.loadFromData(data);
+
+                getApp()->notifications->addNotification(
+                    makeLayout(
+                        avatar,
+                        QString("<b>" + channelName + "</b> just went live!"),
```
Comment (zneix) [inline src/singletons/Toasts.cpp:100]: This just looks cleaner
```suggestion
                        QString("<b>%1</b> just went live!").arg(channelName),
```

File: src/singletons/Toasts.cpp [hunk 4]
```diff
@@ -64,152 +42,108 @@ QString Toasts::findStringFromReaction(
...
+                    callback);
+
+                return Success;
+            })
+        .onError(
+            [channelName, bottomText, callback, this](auto result) -> bool {
+                getApp()->notifications->addNotification(
+                    makeLayout(
+                        getResources().icon,
+                        QString("<b>" + channelName + "</b> just went live!"),
```
Comment (zneix) [inline src/singletons/Toasts.cpp:113]:
```suggestion
                        QString("<b>%1</b> just went live!").arg(channelName),
```

**Review by zneix — CHANGES_REQUESTED** [review]
Apart from some of my minor nitpicks, is there anything stopping this from getting merged? I'd love to see changes from this PR in the application :+1:

**Comment by jammehcow** [comment]
@apa420 what benefit does creating a new notification framework have over fixing how chatterino natively sends notifications?

Maybe I'm missing something.

**Comment by Mm2PL** [comment]
Currently notifications work ONLY on ~~winblows~~ windows

**Comment by apa420** [comment]
Primarily getting the notifications working cross-platform (not only windows version 8 or higher) and removing the wintoast dependancy

**Comment by apa420** [comment]
I'm open to closing this pr if someone gets an easy way of compiling snorenotify on windows. I had so much trouble with it I gave up on it.

**Comment by jammehcow** [comment]
Looks like `snorenotify` has been replaced with [KNotifications](https://api.kde.org/frameworks-api/frameworks-apidocs/frameworks/knotifications/html/index.html). Might be better to pick that up?

**Comment by zneix** [comment]
Related: #2044

I can have a second look at the PR later, update the branch and test it on Linux.

**Comment by pajlada** [comment]
Going ahead and closing this PR, feel free to open a new one or re-open this one if this is something that you're still interested to work on

**Pull request closed without merging**